### ISSUE TITLE
Regression 3.7.2: Global namespace augmentations containing re-exports in nested namespaces not found

### ISSUE THREAD

**Issue by mprobst** [microsoft/TypeScript#35385]
**TypeScript Version:**  Version 3.8.0-dev.20191126

**Search Terms:** namespace, global augmentation, nested

**Code**

```ts
// x.d.ts
export {};

declare class SomeClass {
  method();
}

declare global {
  export namespace myns {
    import SomeName = myns.nested1.SomeName;
    export {SomeName as SomeName2};
  }
  export namespace myns.nested1 {
    export {SomeClass as SomeName};
  }
  export namespace myns.nested2 {
    // export const a: number;
    import SomeName = myns.nested1.SomeName;
    export {SomeName as SomeName2};
  }
}

declare global {  // or just in a separate .d.ts file that's not a module.
  namespace myns.clutz.sth {
    class C extends myns.SomeName2 {}
    class C1 extends myns.nested1.SomeName {}
    class C2 extends myns.nested2.SomeName2 {}
  }
}
```

**Expected behavior:**

Type checks.

**Actual behavior:**

```
x.d.ts:27:27 - error TS2551: Property 'nested2' does not exist on type 'typeof myns'. Did you mean 'nested1'?

27     class C2 extends myns.nested2.SomeName2 {}
                             ~~~~~~~
```

If the namespace *does* contain another export, i.e. if you comment out the `export const a: number`, it is found, including the aliased export.

**Playground Link:** [here](http://www.typescriptlang.org/play/#code/PTAEA8DoBNIFwM4CgCm4AOB7ATnUBvAXwG4kloUBjAGwENsVQbaEFQBlTAWxQGE7WBJKFA84AC0zQAFAEoAXKARxsASwB2Ac1KEyFZg1CbqmAEa1qQkWiy5Q62jwTpalRlwCe6tvmEjQqly2eJw8AHKOjAC8ol4IkOooyijQAIyQoSgRPKT+oDY4ePiZ2YwsHNxZkQBMJH661hiF9pHOru5xCUlwKalW-gV2xZX8LGzlJZF1Ig35TXYOTi5usd5dydDV-SIgc8FMmN54tIrqAK5cpijYuf6B+5M8oDGea4kb6Y8ot437w+GRUATSqlWq3XS6chUOiGYxmCwEHZgHCgABWZ2UAXUQKUKBc2FoPVAMHgbAAZqpqIwJISAORsdSYY6iKRnKmQPyLJLLDprGhnOAAL0gynE2yYAjYvDmPXU0DYr3iXy2RD8ImYgl4fTQsvlq3i7x6aQyIMBqryGqlWx1KDlCs6hpS1RNAJ4KtmEKAA)

**Related Issues:** might be related to #34612.

**Comment by sandersn** [comment]
Looks like to me `nested2` isn't tagged as a value-containing namespace because it contains only imports/exports, even when the import/export aliases a value. I'll have to go looking to figure out the correct way to handle this.

Here's a somewhat simplified repro, although I haven't been able to get rid of the `export { SomeClass as SomeName }`, which I don't really think is required:

```ts
declare class SomeClass {
    method(): any;
}

declare global {
  export namespace myns {
    export { SomeClass as SomeName };
  }
  export namespace myns.nested2 {
    // export const a: number;
    import SomeName = myns.SomeName;
    export { SomeName as SomeName2 };
  }
}

export class C2 extends myns.nested2.SomeName2 {}
```

**Comment by sandersn** [comment]
Fix is up at #37962, although I'm not sure it's efficient or complete enough to ship yet.

**Comment by sandersn** [comment]
A complete, correct fix at #37962 is likely to be quite complicated, so I'm moving this to 4.0.

**Comment by sandersn** [comment]
This is fixed in 5.2, although not by me. I'm pretty sure I heard @andrewbranch working on something similar, though.

**Comment by andrewbranch** [comment]
Oh yeah that’s https://github.com/microsoft/TypeScript/pull/54816
